### PR TITLE
Update regex expressions to handle unexpected '};'

### PR DESCRIPTION
**Pull request by syeopite**
Fixes: #1673
I want to be awarded the bounty associated to the issue this PR is fixing.

Tested with the following ids:
<details>

O4r9D2jI0_w

VQOdmckqNro

3AF3sYnvj5o

BFlF9gGnyro 

BUU7WFd8XLU

channel/UCRZY6MGMG1Fmzt50rYalTwQ

</details>

I wasn't able to reproduce the error in any of the videos linked in duplicate issues. However, this does fix the error in the original video and any other content with "};".

### CODE REVIEW

File: src/invidious/videos.cr [hunk 1]
```diff
@@ -818,7 +818,7 @@ end
 
 def extract_polymer_config(body)
   params = {} of String => JSON::Any
-  player_response = body.match(/(window\["ytInitialPlayerResponse"\]|var\sytInitialPlayerResponse)\s*=\s*(?<info>{.*?});/m)
+  player_response = body.match(/(window\["ytInitialPlayerResponse"\]|var\sytInitialPlayerResponse)\s*=\s*(?<info>{.*?});\s*var\s*meta/m)
```
Comment (SamantazFox) [inline src/invidious/videos.cr:821]: I have noticed that this extra javascript (`var meta = ...`) is not always present in the script.
I can't tell what triggers it (I've tried multiple combinations of URL arguments, with no luck).

Comment (syeopite) [reply src/invidious/videos.cr:821]: Huh that's strange. Its always been present on my end.

**Review by saltycrys — APPROVED** [review]
The most robust way to handle this would be to put a full JSON parser in there so only the starting point needs to be found. That would be terrible for performance probably.

But as long as this works that's not necessary.

**Comment by SamantazFox** [comment]
> The most robust way to handle this would be to put a full JSON parser in there so only the starting point needs to be found. That would be terrible for performance probably.

Or, as proposed in #1981, find a direct API providing the required `ytInitialData` (and `ytInitialPlayerResponse` for videos)...

**Comment by saltycrys** [comment]
> Or, as proposed in #1981, find a direct API providing the required `ytInitialData` (and `ytInitialPlayerResponse` for videos)...

That is fantastic! @TheFrenchGhosty put some bounties on that please.

**Comment by Perflyst** [comment]
`watch?v=VQOdmckqNro` works on my instance without this patch so I can't really test it

**Comment by TheFrenchGhosty** [comment]
Merged.

@syeopite to receive your reward either send your BTC address in a comment (preferred), or contact us directly, with a proof that this is indeed you.

**Comment by syeopite** [comment]
bc1qpgyc5hkcwg4l4a28j274g4tufym07e0as7pwwk

**Comment by TheFrenchGhosty** [comment]
A 20$ bounty has been rewarded to @syeopite for this PR.

Transaction (caa3a36040c086642a0262048f95b0c5d84009e87ddc34acf65b6928895f3fdb):

Amount sent: 0.000405221 BTC (~16.46€ or ~20$)

Mining fees (10 sat/byte): 0.0000481 BTC (~1.95€)

**Pull request merged**